spring-integration-gcp/src/main/java/org/springframework/integration/gcp/storage/outbound/GcsMessageHandler.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.integration.gcp.storage.outbound;
+
+import com.google.cloud.storage.BlobInfo;
+
+import org.springframework.integration.file.remote.RemoteFileTemplate;
+import org.springframework.integration.file.remote.handler.FileTransferringMessageHandler;
+import org.springframework.integration.file.remote.session.SessionFactory;
+import org.springframework.integration.file.support.FileExistsMode;
+
+/**
+ * @author João André Martins
+ */
+public class GcsMessageHandler extends FileTransferringMessageHandler<BlobInfo> {
```

Comment:
I wonder if we can implement `doChmod()` here via that `createAcl(BlobId blob, Acl acl)`.
But that is already different story. Just do not forget that we may have a hook.

Reply:
I'm failing to see a way to map `chmod` permissions to ACLs.
chmod allows the user, its group or everyone to do reads/writes/executes, where ACLs let specific usernames (i.e., Google user accounts), so it doesn't translate very well, but I may not be seeing the whole picture here.

On top of that, I don't really understand why `doChmod()` is a method of `FileTransferringMessageHandler` and not `Session`.

We can leave this as a TODO or open an issue to think about this afterwards?